GitUI/CommandsDialogs/SettingsDialog/Pages/RevisionLinksExtractor.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using GitCommands.ExternalLinks;
+using ResourceManager;
+
+namespace GitUI.CommandsDialogs.SettingsDialog.Pages
+{
+    public enum CloudProviderKind
+    {
+        None = 0,
+        GitHub,
+        AzureDevOps,
+    }
+
+    public interface ICloudProviderExternalLinkDefinitionExtractor
+    {
+        bool IsValidRemoteUrl(string remoteUrl);
+        IList<ExternalLinkDefinition> GetDefinitions(string remoteUrl);
+    }
+
+    public abstract class ExternalLinkDefinitionExtractor : ICloudProviderExternalLinkDefinitionExtractor
+    {
+        protected readonly TranslationString _linkCodeToString = new TranslationString("{0} - Code");
```

Comment:
`private protected readonly TranslationString ...`

Reply:
Sounds locked up to me...